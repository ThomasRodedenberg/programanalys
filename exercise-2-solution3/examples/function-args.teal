fun f(x : nonnull int) = {
    return x; // N 0 0 =
}

fun g(x, y) = {
    return x; // N 0 0 =
}

fun main() = {
    var x : nonnull int := 5; // N 10 27 =, N 10 4 +
    var y := f(x); // NC 11 13 +, N 11 4 = 
    var z := g(x, 5); // NC 12 13 ==, N 12 4 =
    // N 0 0 =
}