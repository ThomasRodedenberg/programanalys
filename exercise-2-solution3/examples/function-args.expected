maybe: N 2 11 +
N 10 4 +
NC 11 13 +
